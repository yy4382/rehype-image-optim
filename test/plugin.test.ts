import rehypeImageOptimization from "../lib";
import { rehype } from "rehype";
import { it, expect, describe } from "vitest";

const input = `<img src="https://example.com/image.jpg">`;
describe("Cloudflare", () => {
  it("should not change src if `optimizeSrcOptions` is undefined", async () => {
    const result = await process(input, { provider: "cloudflare" });
    expect(result).toMatchSnapshot();
  });
  it("should work with default options", async () => {
    const result = await process(input, {
      provider: "cloudflare",
      optimizeSrcOptions: {},
    });
    expect(result).toMatchSnapshot();
  });
  it("should work with srcset", async () => {
    const result = await process(input, {
      provider: "cloudflare",
      srcsetOptionsList: [
        [{}, "1x"],
        [{ options: "w=200" }, "2x"],
      ],
    });
    expect(result).toMatchSnapshot();
  });
  it("should work with sizes", async () => {
    const result = await process(input, {
      provider: "cloudflare",
      sizesOptionsList: ["(max-width: 600px) 100vw", "50vw"],
    });
    expect(result).toMatchSnapshot();
  });
  it("should work with style", async () => {
    const result = await process(input, {
      provider: "cloudflare",
      style: "width: 100%",
    });
    expect(result).toMatchSnapshot();
    const inputWithStyle = `<img src="https://example.com/image.jpg" style="width: 100%;">`;
    expect(
      await process(inputWithStyle, {
        provider: "cloudflare",
        style: "height: auto;",
      }),
    ).toMatchSnapshot();
  });
});

async function process(
  input: string,
  options: Parameters<typeof rehypeImageOptimization>[0],
): Promise<string> {
  const result = await rehype()
    .use({ settings: { fragment: true } })
    .use(rehypeImageOptimization, options)
    .process(input);
  return result.toString();
}
