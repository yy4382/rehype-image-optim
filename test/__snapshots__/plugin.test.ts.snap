// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cloudflare > should not change src if \`optimizeSrcOptions\` is undefined 1`] = `"<img src="https://example.com/image.jpg">"`;

exports[`Cloudflare > should work with default options 1`] = `"<img src="https://example.com/cdn-cgi/image/f=auto/image.jpg">"`;

exports[`Cloudflare > should work with sizes 1`] = `"<img src="https://example.com/image.jpg" sizes="(max-width: 600px) 100vw, 50vw">"`;

exports[`Cloudflare > should work with srcset 1`] = `"<img src="https://example.com/image.jpg" srcset="https://example.com/cdn-cgi/image/f=auto/image.jpg 1x, https://example.com/cdn-cgi/image/w=200/image.jpg 2x">"`;

exports[`Cloudflare > should work with style 1`] = `"<img src="https://example.com/image.jpg" style="width: 100%">"`;

exports[`Cloudflare > should work with style 2`] = `"<img src="https://example.com/image.jpg" style="width: 100%; height: auto;">"`;
